mlflow/store/artifact/runs_artifact_repo.py

```diff
@@ -33,22 +33,28 @@ def __init__(self, artifact_uri: str, tracking_uri: Optional[str] = None) -> Non
         from mlflow.store.artifact.artifact_repository_registry import get_artifact_repository
 
         super().__init__(artifact_uri, tracking_uri)
-        uri = RunsArtifactRepository.get_underlying_uri(artifact_uri)
-        self.repo = get_artifact_repository(uri)
+        uri = RunsArtifactRepository.get_underlying_uri(artifact_uri, tracking_uri)
+        self.repo = get_artifact_repository(uri, self.tracking_uri)
 
     @staticmethod
     def is_runs_uri(uri):
         return urllib.parse.urlparse(uri).scheme == "runs"
 
     @staticmethod
-    def get_underlying_uri(runs_uri):
+    def get_underlying_uri(runs_uri: str, tracking_uri: Optional[str] = None) -> str:
         from mlflow.tracking.artifact_utils import get_artifact_uri
 
         (run_id, artifact_path) = RunsArtifactRepository.parse_runs_uri(runs_uri)
-        tracking_uri = get_databricks_profile_uri_from_artifact_uri(runs_uri)
-        uri = get_artifact_uri(run_id, artifact_path, tracking_uri)
+        tracking_uri_from_run = get_databricks_profile_uri_from_artifact_uri(runs_uri)
+        uri = get_artifact_uri(
+            run_id=run_id,
+            artifact_path=artifact_path,
+            tracking_uri=tracking_uri_from_run or tracking_uri,
```

Comment:
What if both tracking_uri_from_run and tracking_uri exist? 🤔

Reply:
Good point. To be conservative, I chose to use the return value of `get_databricks_profile_uri_from_artifact_uri` to avoid behavior changes.